src/gc.c

```diff
@@ -519,17 +577,24 @@ static inline uint16_t gc_setmark_pool_(jl_ptls_t ptls, jl_taggedvalue_t *o,
     uint16_t tag_changed = !gc_marked(tag);
     if (tag_changed) {
         if (mark_mode == GC_OLD_MARKED) {
-            perm_scanned_bytes += page->osize;
-            page->nold++;
+            ptls->gc_cache.perm_scanned_bytes += page->osize;
+            jl_atomic_fetch_add_relaxed(&page->nold, 1);
```

Comment:
should we try to rethink how we detect `nold == prev_nold` to avoid the atomic here?